apps/dashboard/src/components/workflow-editor/steps/step-skeleton.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { StepTypeEnum } from '@novu/shared';
+import { RiEdit2Line, RiPencilRuler2Line } from 'react-icons/ri';
+import { useNavigate } from 'react-router-dom';
+import { Cross2Icon } from '@radix-ui/react-icons';
+
+import { Skeleton } from '@/components/primitives/skeleton';
+import { Button } from '@/components/primitives/button';
+import { Separator } from '@/components/primitives/separator';
+import { Tabs, TabsList, TabsTrigger } from '@/components/primitives/tabs';
+import { Notification5Fill } from '@/components/icons';
+import { WorkflowOriginEnum } from '@/utils/enums';
+
+const SingleLineSkeleton = () => {
+  return (
+    <div className="flex h-full flex-col gap-1">
+      <Skeleton className="h-10 w-full" />
+    </div>
+  );
+};
+
+const STEP_TYPE_TO_SKELETON_CONTENT: Record<StepTypeEnum | string, () => React.JSX.Element | null> = {
+  [StepTypeEnum.EMAIL]: () => {
+    return (
+      <div className="flex h-full flex-col gap-1">
+        <Skeleton className="h-10 w-full" />
+        <Skeleton className="h-full w-full" />
+      </div>
+    );
+  },
+  [StepTypeEnum.CHAT]: SingleLineSkeleton,
+  [StepTypeEnum.IN_APP]: () => {
+    return (
+      <>
+        <div className="flex items-center gap-2.5">
+          <RiPencilRuler2Line className="text-foreground-950 size-5 p-0.5 text-sm font-medium" />
+          <span>In-app Template</span>
+        </div>
+        <div className="flex flex-col gap-1 rounded-xl border border-neutral-100 p-1">
+          <div className="flex gap-1">
+            <Skeleton className="h-10 min-w-10" />
+            <Skeleton className="h-10 w-full" />
+          </div>
+          <Skeleton className="h-32 w-full" />
+        </div>
+      </>
+    );
+  },
+  [StepTypeEnum.SMS]: SingleLineSkeleton,
+  [StepTypeEnum.PUSH]: SingleLineSkeleton,
+  [StepTypeEnum.DIGEST]: () => null,
+  [StepTypeEnum.DELAY]: () => null,
+  [StepTypeEnum.TRIGGER]: () => null,
+  [StepTypeEnum.CUSTOM]: () => null,
+};
+
+export const StepSkeleton = ({
```

Comment:
the step skeleton which simulates the step editor shell for the sidebar with: header, body, footer